[WebGPU] webgpu:api,operation,reflection:* does not pass

mwyrzykowski opened the pull request:
#### 3dbba0e01464ac78587324bd14314c6945c4f821
<pre>
[WebGPU] webgpu:api,operation,reflection:* does not pass
<a href="https://bugs.webkit.org/show_bug.cgi?id=264547">https://bugs.webkit.org/show_bug.cgi?id=264547</a>
&lt;radar://118216777&gt;

Reviewed by Tadeu Zagallo.

Fix a validation error in texture and query set creation and
update the GPUQuerySet idl to the most recent version.

* Source/WebCore/Modules/WebGPU/GPUDevice.cpp:
(WebCore::GPUDevice::createQuerySet):
* Source/WebCore/Modules/WebGPU/GPUQuerySet.cpp:
(WebCore::GPUQuerySet::GPUQuerySet):
(WebCore::GPUQuerySet::type const):
(WebCore::GPUQuerySet::count const):
* Source/WebCore/Modules/WebGPU/GPUQuerySet.h:
(WebCore::GPUQuerySet::create):
(WebCore::GPUQuerySet::GPUQuerySet): Deleted.
* Source/WebGPU/WebGPU/QuerySet.mm:
(WebGPU::Device::createQuerySet):
* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::Device::validateCreateTexture):
* Source/WebCore/Modules/WebGPU/GPUQuerySet.idl:

Canonical link: <a href="https://commits.webkit.org/270577@main">https://commits.webkit.org/270577@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c8bed0da9ad1a10f61a75d13fd726b6698d4aa46

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/25787 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/4394 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27071 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/27885 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/23612 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/26072 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/6155 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/1830 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/23719 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/26037 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3301 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22220 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/28465 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/2923 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/23176 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/29242 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/23535 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/23547 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27111 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/2941 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1173 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/4328 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6208 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/3394 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3256 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski commented on Source/WebGPU/WebGPU/QuerySet.mm:
```diff
@@ -34,11 +34,12 @@
 {
-    if (descriptor.nextInChain)
+    auto count = descriptor.count;
+    if (descriptor.nextInChain || count > 4096 || !isValid()) {
```
4096 comes from the WebGPU specification

tadeuzagallo replied:
maybe add a comment or a constant?

mwyrzykowski committed:
[WebGPU] webgpu:api,operation,reflection:* does not pass
https://bugs.webkit.org/show_bug.cgi?id=264547
<radar://118216777>

Reviewed by Tadeu Zagallo.

Fix a validation error in texture and query set creation and
update the GPUQuerySet idl to the most recent version.

* Source/WebCore/Modules/WebGPU/GPUDevice.cpp:
(WebCore::GPUDevice::createQuerySet):
* Source/WebCore/Modules/WebGPU/GPUQuerySet.cpp:
(WebCore::GPUQuerySet::GPUQuerySet):
(WebCore::GPUQuerySet::type const):
(WebCore::GPUQuerySet::count const):
* Source/WebCore/Modules/WebGPU/GPUQuerySet.h:
(WebCore::GPUQuerySet::create):
(WebCore::GPUQuerySet::GPUQuerySet): Deleted.
* Source/WebGPU/WebGPU/QuerySet.mm:
(WebGPU::Device::createQuerySet):
* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::Device::validateCreateTexture):
* Source/WebCore/Modules/WebGPU/GPUQuerySet.idl:

Canonical link: https://commits.webkit.org/270577@main
